### PR TITLE
Add service update test in lifecycle test

### PR DESCRIPTION
**Pull request by zhongyi-zhang**
Fixed https://github.com/Azure/open-service-broker-azure/issues/465.

Also fixed the updating behavior of mssql database-only service.

### CODE REVIEW

File: pkg/services/mssql/database_update.go [hunk 1]
```diff
@@ -27,31 +27,29 @@ func (d *databaseManager) updateARMTemplate(
 	instance service.Instance,
 ) (service.InstanceDetails, error) {
 	dt := instance.Details.(*databaseInstanceDetails)
-	pd, ok := instance.Plan.GetProperties().Extended["tierDetails"]
-	if !ok {
-		return nil, fmt.Errorf("unable to access plan details")
-	}
-	planDetails, _ := pd.(planDetails)
+	pdt := instance.Parent.Details.(*dbmsInstanceDetails)
+	pd := instance.Plan.GetProperties().Extended["tierDetails"].(planDetails)
```
Comment (zhongyi-zhang) [inline pkg/services/mssql/database_update.go]: Here follows the style of database_provision.go.

**Comment by krancour** [comment]
@zhongyi-zhang this looks excellent! Thank you so much. Will merge pending CI.

**Comment by krancour** [comment]
@zhongyi-zhang there were some lint failures.

**Comment by krancour** [comment]
Thanks @zhongyi-zhang!

**Pull request merged**